src/box/iproto.cc

```diff
@@ -3038,36 +3089,75 @@ iproto_free(void)
 	evio_service_stop(&tx_binary);
 }
 
-int
-iproto_rmean_foreach(void *cb, void *cb_ctx)
+static int
+iproto_thread_rmean_foreach_impl(struct rmean *rmean, void *cb, void *cb_ctx)
 {
-	for (size_t i = 0; i < RMEAN_NET_LAST; i++) {
+	int rc = 0;
+	for (size_t i = 0; i < rmean->stats_n; i++) {
+		int64_t mean = rmean_mean(rmean, i);
+		int64_t total = rmean_total(rmean, i);
+		if (((rmean_cb)cb)(rmean->stats[i].name, mean,
+				   total, cb_ctx) != 0)
+			rc = 1;
+	}
+	return rc;
+}
+
+/**
+ * We use offset of rmean in struct iproto_thread, instead of pointer to
+ * rmean, because we should iterate over all same rmeans for all iproto
+ * threads.
+ */
+static int
+iproto_rmean_foreach_impl(ptrdiff_t rmean_offset, void *cb, void *cb_ctx)
+{
+	struct rmean **sample_rmean =
+		(struct rmean **)((char *)&iproto_threads[0] + rmean_offset);
```

Comment:
Nit: I'd call it rmean0, because it's not a sample per se, and dereference it right here rather than in every place where it's accessed:

```c
struct rmean *rmean0 = *(struct rmean **)((char *)&iproto_threads[0] + rmean_offset);
for (size_t i = 0; i < rmean0->stats_n; i++) {
```

Same comment for the rmean below.

Reply:
fixed